middleware_auth_user.go

```diff
@@ -47,13 +42,8 @@ func (umc *UserMiddlewareConfiguration) Validate() error {
 	}
 
 	if !umc.SkipHeaderCheck {
-		if umc.HeaderUserName == "" {
-			catcher.New("when header auth is enabled, must specify a header user name")
-		}
-
-		if umc.HeaderKeyName == "" {
-			catcher.New("when header auth is enabled, must specify a header key name")
-		}
+		catcher.NewWhen(umc.HeaderUserName == "", "when header auth is enabled, must specify a header user name")
```

Comment:
Can we change this and the one below to specify the error first then the condition?
- `must specify a header user name when header auth is enabled`
- `must specify a header key name when header auth is enabled`

Reply:
Done.